scripts/py_matter_yamltests/matter_yamltests/parser.py

```diff
@@ -565,27 +635,35 @@ def _skip_post_processing(self, response: dict, result) -> bool:
         if not self.optional:
             return False
 
-        received_error = response.get('error', None)
+        check_type = PostProcessCheckType.OPTIONAL
+        error_failure_multiple_responses = 'The test expects a single response but got {len(received_responses)} responses.'
+
+        if len(received_responses) > 1:
+            result.error(check_type, error_failure.multiple_responses)
+            return False
+        received_response = received_responses[0]
+
+        received_error = received_response.get('error', None)
         if received_error is None:
             return False
 
         if received_error == 'UNSUPPORTED_ATTRIBUTE' or received_error == 'UNSUPPORTED_COMMAND':
-            # result.warning(PostProcessCheckType.Optional, f'The response contains the error: "{error}".')
+            # result.warning(PostProcessCheckType.OPTIONAL, f'The response contains the error: "{error}".')
```

Comment:
Why do we have this commented-out code here?
Could a comment explain it or maybe delete it alltogether?

Reply:
I have uncommented it.